src/bot/BotController.ts

```diff
@@ -31,6 +33,18 @@ export class BotController {
       }
     });
 
+    this.client.on('message', (msg: Message) => {
+      const channel = msg.channel as TextChannel;
```

Comment:
We can't be sure its from a TextChannel so instead of forcing the type here we should check if `msg.channel` is an instance of `TextChannel` and ignore it if its not.

Reply:
Point, checking that now.